### DISCUSSION TITLE
请教一个关于多个pag文件合成一个画布的问题

### DISCUSSION THREAD

**Discussion by hubeishuaige** [Tencent/libpag#1958] [Q&A] [answered]
各位大佬，我是新手，最近公司着迷，迷上了libpag.js ，作为新手，我不得不来请教一些问题，望解答，本人跪谢：
1.多个pag文件，每个pag文件里面都有文本和图片，当我把他们合成到一个画布上的时候，怎么修改对应的文字或者图片？比如说，我点击其中的一个图片，我希望能修改动态修改图片，但是我怎么知道我当前点击的是哪个对象？我希望我在点击某个图层的时候，能够给这个图层添加一个border，但是不知道怎么实现？

2.关于没有办法使用上下文，我是在JS里面创建了一个画布，然后插入到body里面，但是感觉这种写法比较不友好，不知道有没有更好i的方案。

3.拖拽和旋转某个图层，至今不知道怎么使用？有谁知道，可以告诉一下吗？跪谢

附上图片demo，希望大家可以指点一下，非常感谢，最后再问一下，有没有沟通的QQ群或者微信群，麻烦留个交流群的联系方式
![pag-demo](https://github.com/Tencent/libpag/assets/21007762/a5ec3194-7e64-4a84-9d1c-de27b0edd539)

**Comment by domchen** [comment, marked as answer]
1.使用 PAGComposition.getLayersUnderPoint()可以获取点击坐标下方的图层列表，你遍历返回的数组看类型是不是可以编辑的，命中的第一个图层就是最上面的，直接操作这个图层对应的接口进行编辑修改即可。点击的时候添加边框这个不是pag的能力，但是你可以把前面点击获取的图层调用PAGPlayer.getBounds(PAGLayer)获取像素边界，然后自己绘制一个边框叠加上去。

2.这部分是业务逻辑，你自己选择合适的方式都可以。
3.你是想改变图层的位置或者旋转缩放，通过PAGLayer.setMatrix()就行。Matrix是标准的2x3仿射变换矩阵，可以表示一切旋转缩放位移，要是不知道怎么使用可以谷歌一下其他平台的仿射变换矩阵。
